internal/project/project.go

```diff
@@ -205,6 +235,95 @@ func (p *Project) LanguageService() *ls.LanguageService {
 	return p.languageService
 }
 
+func (p *Project) getRootFileWatchGlobs() []string {
+	if p.kind == KindConfigured {
+		wildcardDirectories := p.parsedCommandLine.WildcardDirectories()
+		result := make([]string, 0, len(wildcardDirectories)+1)
+		result = append(result, p.configFileName)
+		for dir, recursive := range wildcardDirectories {
+			result = append(result, fmt.Sprintf("%s/%s", dir, core.IfElse(recursive, recursiveFileGlobPattern, fileGlobPattern)))
+		}
+		return result
+	}
+	return nil
+}
+
+func (p *Project) getModuleResolutionWatchGlobs() (failedLookups map[tspath.Path]string, affectingLocaions map[tspath.Path]string) {
+	failedLookups = make(map[tspath.Path]string)
```

Comment:
This may be ok for initial implementation but iterating over so many resolutions and so many failed lookups has been perf bottleneck. It was also memory issue for bigger code bases to store all the failed lookups as is and not watch the directories instead

Reply:
Yeah, I want to do the simplest thing to start with and then optimize based on measurements